### PR TITLE
Docs: Clarify docs for Overload Manager disable_keep_alive action.

### PR DESCRIPTION
**Pull request by KBaichoo**
Signed-off-by: Kevin Baichoo <kbaichoo@google.com>

Commit Message:  Clarify docs for Overload Manager disable_keep_alive action.
Additional Description:

Looking at the code: https://github.com/envoyproxy/envoy/blob/0a4e3d990ae865c6753432322e9b2eaec8057548/source/common/http/conn_manager_impl.cc#L1382-L1401 

We seem to use GoAways for H2/H3 and a drain timer for H/1 which will close the connection if it has no stream and nothing to write when the timer fires.

Risk Level: NA
Testing: NA
Docs Changes: Included
Release Notes: NA
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: docs/root/configuration/operations/overload_manager/overload_manager.rst [hunk 1]
```diff
@@ -81,7 +81,9 @@ The following overload actions are supported:
     - Envoy will immediately respond with a 503 response code to new requests
 
   * - envoy.overload_actions.disable_http_keepalive
-    - Envoy will stop accepting streams on incoming HTTP connections
+    - Envoy will drain HTTP/2, HTTP/3 connections using `GOAWAY` with a drain
```
Comment (phlax) [inline docs/root/configuration/operations/overload_manager/overload_manager.rst:84]: hi @KBaichoo can you use double backticks for rst please ` `` ` - single backtick creates italics, whereas we want an inline literal - you may want to consider likewise for some of the other literals - eg `HTTP/2` etc but maybe not

**Comment by phlax** [comment]
docs are rendered here https://storage.googleapis.com/envoy-pr/19558/docs/index.html

specifically here on this commit https://storage.googleapis.com/envoy-pr/1e1582c/docs/configuration/operations/overload_manager/overload_manager.html#overload-actions

File: docs/root/configuration/operations/overload_manager/overload_manager.rst (hunk 1 above)
Comment (KBaichoo) [reply docs/root/configuration/operations/overload_manager/overload_manager.rst]: Whoops, thanks for catching that. GOAWAY is now inline literal

File: docs/root/configuration/operations/overload_manager/overload_manager.rst [hunk 2]
```diff
@@ -81,7 +81,9 @@ The following overload actions are supported:
     - Envoy will immediately respond with a 503 response code to new requests
 
   * - envoy.overload_actions.disable_http_keepalive
-    - Envoy will stop accepting streams on incoming HTTP connections
+    - Envoy will drain HTTP/2, HTTP/3 connections using ``GOAWAY`` with a drain
```
Comment (daixiang0) [inline docs/root/configuration/operations/overload_manager/overload_manager.rst:84]: HTTP/2 and HTTP/3 here?

Comment (KBaichoo) [reply docs/root/configuration/operations/overload_manager/overload_manager.rst]: done.

**Review by phlax — APPROVED** [review]
lgtm, thanks @KBaichoo

**Pull request merged**